### ISSUE TITLE
Issues with sending binary data attachment

### ISSUE THREAD

**Issue by rdelpeso** [EC-Nordbund/denomailer#25]
Hi, I am likely doing something wrong. I love the lib but I can't figure out how to send a PNG attachment with it. This is what am doing, any suggestions?

```ts
const file = await Deno.readFile("./test.png");
await client.send({
  attachments: [
    {
      contentType: "image/png",
      filename: "test.png",
      encoding: "binary",
      content: file,
    },
  ],
  from: "address@gmail.com",
  to: "address@gmail.com",
  subject: "Img Test",
  content: "",
  html:
    "<b>Testing with html and image.</b>",
});
```

When I compare the original PNG and the Sent one. This line is extra on the top of the binary blob:

original.png
```
‰PNG
```

received.png
```
Content-Transfer-Encoding: binary
‰PNG
```

Bonus: Could we add "Content-ID" to the library to support embedding the image on the html? 😄

**Comment by mathe42** [comment]
Not sure why this happens. Currently not home so will take a week at least to look into it.

**Comment by rdelpeso** [comment]
Appreciate it. I will take a stab at it this weekend and submit a pull request if I figure it out. Was just worried I was being blind and missing something 👍

**Comment by mathe42** [comment]
For the Feature request:

I'm currently in a phase where I will not implement new 
features as I don't have (currently!) the time for that. But I would except a PR.

**Comment by mathe42** [comment]
https://github.com/EC-Nordbund/denomailer/blob/7da737e17c989695de9ec8788de66f8ed5889035/client/basic/client.ts#L263

I think this Line might be the problem could you try it when there is no \r\n?

**Comment by rdelpeso** [comment]
Sadly no matter what I did, Gmail refuses to receive the binary attachment.

I've forked the repo, and made some changes to allow for base64 attachments and added some defaults for X-Attachment-Id and Content-ID. Without those, Gmail also refused to work for base64.

Not sure if the changes are up to your taste. Take a look at them when you can, and let me know if you want me to create a PR to make it official.

https://github.com/rdelpeso/denomailer/commit/a39bada309f471ff287dc8ea2e5b532878334499

Now this example works perfectly:

```typescript
const file = await Deno.readFile("./test.png");

import { encode } from "https://deno.land/std/encoding/base64.ts";
const b64 = encode(file);

await client.send({
  attachments: [
    {
      contentType: "image/png",
      filename: "test_64.png",
      encoding: "base64",
      content: b64,
    },
  ],
  from: "address@gmail.com",
  to: "address@gmail.com",
  subject: "Img Test",
  content: "",
  html:
    // cid:attachment_id_0 -> just means the 0th (or first) attachment, this supports embedding now 😄 
    "<b>Testing with html and image.</b><br/><img src="cid:attachment_id_0"/>",
});
```

**Comment by rdelpeso** [comment]
So apparently the fix only worked for Gmail, hotmail is still busted and regardless of using base64 or binary, the attachments are corrupted on the receiving side. haiz...

**Comment by mathe42** [comment]
Will look at it in friday currently not home...

**Comment by rdelpeso** [comment]
Also notice this:

Original File:
![image](https://user-images.githubusercontent.com/1075376/172912957-59694730-d161-42a5-bbdf-1a6b886335c8.png)


After email sent file:
![image](https://user-images.githubusercontent.com/1075376/172913052-82c66508-20f2-4c78-a166-ca49e52ea147.png)


So, all \r are being transformed to \r\n
and all \n are being transformed to \r\n

Which corrupts the image entirely. Either writer is doing that, or GMAIL SMTP is doing it.

**Comment by mathe42** [comment]
Thanks for testing!

**Comment by mathe42** [comment]
I could reproduce it with the same image and gmail i get problems with gmx (a german provider) I have no problems.

**Comment by mathe42** [comment]
Got it...

content-encoding isn't valid... You have to use BINARYMIME. That means that I will rewrite it so that (for now) attachments will be base64 encoded and will later add support for BINARYMIME as I have to read and learn some stuff for that.

Refrences
1. https://stackoverflow.com/a/28531705
2. https://datatracker.ietf.org/doc/html/rfc3030

**Comment by mathe42** [comment]
Released 1.2 with a fix
